src/server/routes/consents.ts

```diff
@@ -279,19 +294,26 @@ router.get(
       const { read, pageTitle: _pageTitle } = consentPages[pageIndex];
       pageTitle = _pageTitle;
 
-      res.locals.pageData = await read(
-        req.ip,
-        sc_gu_u,
-        res.locals.pageData.geolocation,
-      );
-      res.locals.pageData.returnUrl = res.locals?.queryParams?.returnUrl;
+      state = {
+        ...state,
+        pageData: {
+          ...(await read(req.ip, sc_gu_u, state.pageData.geolocation)),
+          returnUrl: state?.queryParams?.returnUrl,
+        },
+      };
     } catch (e) {
       status = e.status;
-      res.locals.globalMessage.error = e.message;
+      state = {
+        ...state,
+        globalMessage: {
+          ...state.globalMessage,
+          error: e.message,
```

Comment:
unrelated to immutability - but I don't think we can be certain that `e.message`  is a message for the user

Reply:
Quite possible, but outside the scope of this PR

Reply:
Looks like Gateway has a pattern of using errors such as
```
 throw { message: ConsentsErrors.GENERIC, status: 500 };
```
in business logic which isn't great. I could raise a ticket to refactor this out.